benches/src/main.rs

```diff
@@ -10,9 +12,11 @@ async fn main() -> eyre::Result<()> {
     );
 
     let reports = [reports.0?, reports.1?, reports.2?, reports.3?];
```

Comment:
This is a nit, but I don't love the idea of accessing the values of the tuple by index(hard to scale and avoid missing out some of the values). Even if it is more verbose this is better IMO and would allow the compiler to highlight when there's an unused value.
```
    let (access_control_report, erc20_report, erc721_report, merkle_proofs_report) = tokio::join!(
        access_control::bench(),
        erc20::bench(),
        erc721::bench(),
        merkle_proofs::bench()
    );

    let reports = [access_control_report?, erc20_report?, erc721_report?, merkle_proofs_report?];
```

Reply:
If you're curious, there is a functional and a bit overengineered solution:
```rust
    let report = futures::future::try_join_all([
        access_control::bench().boxed(),
        erc20::bench().boxed(),
        erc721::bench().boxed(),
        merkle_proofs::bench().boxed(),
    ])
    .await?
    .into_iter()
    .fold(BenchmarkReport::default(), BenchmarkReport::merge_with);
```
I can change to this

Reply:
I like it! :rocket: